### ISSUE TITLE
Deprecation warnings for decision boundary plots

### ISSUE THREAD

**Issue by lesteve** [INRIA/scikit-learn-mooc#495]
```
/opt/hostedtoolcache/Python/3.7.12/x64/lib/python3.7/site-packages/sklearn/base.py:446: UserWarning: X does not have valid feature names, but LogisticRegression was fitted with feature names
  "X does not have valid feature names, but"
```
This seems to affect all the decision boundary plots e.g. this one (https://inria.github.io/scikit-learn-mooc/python_scripts/trees_classification.html):
![image](https://user-images.githubusercontent.com/1680079/144402549-3fb7fde6-522a-4592-9c92-69dea69d019c.png)

**Comment by ArturoAmorQ** [comment]
Should be fixed by merging #484

**Comment by glemaitre** [comment]
Done
